cmd/deploy/deploy.go

```diff
@@ -618,3 +595,82 @@ func (dc *DeployCommand) cleanUp(ctx context.Context) {
 		oktetoLog.Infof("could not stop local server: %s", err)
 	}
 }
+
+func (dc *DeployCommand) buildImages(ctx context.Context, cfg *corev1.ConfigMap, c kubernetes.Interface, pipelineData *pipeline.CfgData, deployOptions *Options) error {
+	if deployOptions.Build {
+		buildOptions := &types.BuildOptions{
+			EnableStages: true,
+			Manifest:     deployOptions.Manifest,
+			CommandArgs:  deployOptions.servicesToDeploy,
+		}
+		oktetoLog.Debug("force build from manifest definition")
+		if errBuild := dc.Builder.Build(ctx, buildOptions); errBuild != nil {
+			return updateConfigMapStatusError(ctx, cfg, c, pipelineData, errBuild)
```

Comment:
I tried that, but doing it after the function changes the behavior since in the `else` we are calling it conditionally (only if there are more than 0 `servicesToBuild`). 

Another approach to reduce the amount of parameters is to take as a parameter the `updateFunction`

Reply:
why don't we return `dc.Builder.Build`, and with that result of `dc.buildImages` we then `updateConfigMapStatusError` ?? this way i think is clearer first is build then is update, no need to have this func inside buildImages ?? WDYT?

Reply:
@teresaromero that's a great idea! I'll do it rn